src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,112 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useCallback, useState } from 'react';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  addNewComment: (comment: Comment) => Promise<void>;
+  isCommentSending: boolean;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  addNewComment,
+  isCommentSending,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [isSubmit, setIsSubmit] = useState(false);
+
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isTextError, setIsTextError] = useState(false);
+
+  const validateEmail = useCallback((emailInput: string) => {
+    const reg = /^[A-Za-z0-9_!#$%&'*+?`{|}~^.-]+@[A-Za-z0-9.-]+$/gm;
+
+    return reg.test(emailInput);
+  }, [email]);
+
+  const checkFormData = useCallback(() => {
+    if (!name.trim()) {
+      setName('');
+      setIsNameError(true);
+    }
+
+    if (!email.trim()) {
+      setEmail('');
+      setIsEmailError(true);
+    }
+
+    if (!text.trim()) {
+      setText('');
+      setIsTextError(true);
+    }
+  }, [name, email, text]);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setIsSubmit(true);
+    checkFormData();
+
+    if (!(name.trim() && validateEmail(email) && text.trim())) {
+      return;
+    }
+
+    const newComment: Comment = {
+      id: 0,
+      postId,
+      name,
+      email,
+      body: text,
+    };
+
+    addNewComment(newComment);
+    setText('');
+    setIsSubmit(false);
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const inputName = e.currentTarget.name;
+    const inputValue = e.currentTarget.value;
+
+    switch (inputName) {
+      case 'name':
```

Comment:
create enum for this
```suggestion
      case SomeEnum.Name:
```